Upstream-Status: Pending

Adjust common software manager code for the updated sdbusplus API.

diff --git a/common/include/software_update.hpp b/common/include/software_update.hpp
--- a/common/include/software_update.hpp
+++ b/common/include/software_update.hpp
@@ -29,7 +29,8 @@

     ~SoftwareUpdate();

-    auto method_call(start_update_t su, auto image, auto applyTime)
+    auto method_call(start_update_t su, auto image, auto applyTime,
+                     auto forceUpdate, auto targets)
         -> sdbusplus::async::task<start_update_t::return_type>;

     auto get_property(allowed_apply_times_t aat) const;
diff --git a/common/src/software_update.cpp b/common/src/software_update.cpp
--- a/common/src/software_update.cpp
+++ b/common/src/software_update.cpp
@@ -38,9 +38,12 @@
 }

 auto SoftwareUpdate::method_call(start_update_t /*unused*/, auto image,
-                                 auto applyTime)
+                                 auto applyTime, auto forceUpdate,
+                                 auto targets)
     -> sdbusplus::async::task<start_update_t::return_type>
 {
+    (void)forceUpdate;
+    (void)targets;
     debug("Requesting Image update with {FD}", "FD", image.fd);

     Device& device = software.parentDevice;
diff --git a/common/src/software_manager.cpp b/common/src/software_manager.cpp
--- a/common/src/software_manager.cpp
+++ b/common/src/software_manager.cpp
@@ -32,8 +32,7 @@
     configIntfAddedMatch(ctx, RulesIntf::interfacesAdded() + matchRuleSender),
     configIntfRemovedMatch(ctx, RulesIntf::interfacesRemoved() + matchRulePath),
     serviceName("xyz.openbmc_project.Software." + serviceNameSuffix),
-    manager(ctx, sdbusplus::client::xyz::openbmc_project::software::Version<>::
-                     namespace_path)
+    manager(ctx, "/xyz/openbmc_project/software")
 {
     debug("requesting dbus name {BUSNAME}", "BUSNAME", serviceName);
